pkg/store/consul/dsstore/consul_store.go

```diff
@@ -40,18 +46,18 @@ func (e CASError) Error() string {
 	return fmt.Sprintf("Could not check-and-set key %q", string(e))
 }
 
-// Make sure functions declared in the Store interface have the proper contract
-var _ Store = &consulStore{}
-
-func NewConsul(client consulutil.ConsulClient, retries int, logger *logging.Logger) Store {
-	return &consulStore{
+func NewConsul(client consulutil.ConsulClient, retries int, logger *logging.Logger) *store {
```

Comment:
Basically i want this type to return a type that has the full range of capability of talking to consul as it relates to daemon sets, but if it is captured into small interface variables where it is used then it will be easier to swap implementations